### PR TITLE
Fix lack of clippy on nix

### PR DESCRIPTION
**Pull request by nixos-goddess**
Overrode the rust package so it'll install clippy

### CODE REVIEW

**Comment by mfdorst** [comment]
This is awesome, thank you. I did some testing and you don't need the override, so it can just be:
```nix
            nativeBuildInputs = [ pkgs.rust-bin."${rust-toolchain}".latest.default ];
```
Also, I don't believe this part of `magmawm.nix` is doing anything:
```nix
  rust = pkgs.rust-bin."${rust-toolchain}".latest.default.override {
    extensions = [
      "cargo"
      "clippy"
      "rust-src"
      "rustc"
    ];
  };
```
I think it can just be removed.

**Comment by nixos-goddess** [comment]
Oh
Alr :p
Ty
I hate coding from my phone

**Comment by mfdorst** [comment]
I can do it for you if you like

File: magmawm.nix [hunk 1]
```diff
@@ -10,7 +10,14 @@ pkgs.rustPlatform.buildRustPackage {
...
 
-  rust = pkgs.rust-bin."${rust-toolchain}".latest.default;
+  rust = pkgs.rust-bin."${rust-toolchain}".latest.default.override {
+    extensions = [
+      "cargo"
+      "clippy"
+      "rust-src"
+      "rustc"
+    ];
+  };
```
[the remark this answers is not in the document]
Comment (mfdorst) [reply magmawm.nix]: Resolved

File: flake.nix [hunk 2]
```diff
@@ -25,7 +25,7 @@
       overlays.default = final: _prev:
         let
           version = (builtins.fromTOML (builtins.readFile ./Cargo.toml)).package.version;
-          rust-toolchain = "stable";
```
[the remark this answers is not in the document]
Comment (mfdorst) [reply flake.nix:28]: This isn't that important, we can merge without it.

**Review by mfdorst — APPROVED** [review]

File: flake.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (nixos-goddess) [reply flake.nix:28]: Yeah, it's annoying me too

**Review by electron271 — APPROVED** [review]

**Pull request merged**